pallets/creator-staking/src/inflation.rs

```diff
@@ -0,0 +1,44 @@
+use crate::*;
+use frame_support::traits::{Currency, Get, OnTimestampSet, Imbalance};
+use sp_runtime::traits::{Saturating, SaturatedConversion, UniqueSaturatedInto, Zero};
+
+type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+impl<Moment, T: Config> OnTimestampSet<Moment> for Pallet<T> {
+    fn on_timestamp_set(_moment: Moment) {
+        let total_issuance = T::Currency::total_issuance();
+        let reward_amount: BalanceOf<T> = T::CurrentAnnualInflation::get() * total_issuance
+            / T::BlocksPerYear::get().saturated_into::<u32>().unique_saturated_into();
+
+        let inflation = T::Currency::issue(reward_amount);
+        Self::distribute_rewards(inflation);
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    fn distribute_rewards(block_reward: NegativeImbalanceOf<T>) {
```

Comment:
What do we do then?

---

The given function is responsible for distributing a block reward among various beneficiaries. Here's a breakdown of what the function does:

1. Fetch the distribution parameters (`distro_params`) from some configuration.
2. Calculate the amounts to be rewarded to stakers, creators, and the treasury based on the provided `block_reward`.
3. Split the `block_reward` into imbalances for creators, stakers, and treasury, according to the pre-calculated balances.
4. Update `BlockRewardAccumulator` by adding the imbalances for creators and stakers to the accumulated rewards for those groups.
5. Use the `resolve_creating` function to actually distribute the rewards to the beneficiaries' accounts (`Self::rewards_pot_account()` for creators and stakers, and `T::TreasuryAccount::get()` for the treasury).

As for the name, `distribute_rewards` is already quite descriptive. However, if you want it to be even more specific, you could rename it to something like `distribute_block_rewards` or `allocate_block_rewards_to_beneficiaries`. This would make it clear that the rewards being distributed are specifically related to a block and go to multiple types of beneficiaries (creators, stakers, and the treasury).

Reply:
Give a name something like: Mint accumulated  rewards in this era.